### PR TITLE
Implement akita maps

### PR DESCRIPTION
**Pull request by sei-byoung**

### CODE REVIEW

File: src/app/state/vmMaps/vm-maps.query.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { Injectable } from '@angular/core';
```
Comment (sei-tspencer) [inline src/app/state/vmMaps/vm-maps.query.ts:1]: Header

File: src/app/state/vmMaps/vm-maps.service.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { HttpClient } from '@angular/common/http';
```
Comment (sei-tspencer) [inline src/app/state/vmMaps/vm-maps.service.ts:1]: Header

File: src/app/state/vmMaps/vm-maps.store.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { Injectable } from '@angular/core';
```
Comment (sei-tspencer) [inline src/app/state/vmMaps/vm-maps.store.ts:1]: header

**Review by sei-tspencer — APPROVED** [review]

**Comment by sei-ebram** [comment]
there's a bunch of LGTM alerts that need to be fixed as well

File: src/app/components/map/add-point/add-point.component.ts [hunk 4]
```diff
@@ -28,67 +34,97 @@ export class AddPointComponent implements OnInit {
...
+    this.vmMapsFiltered = this.route.params.pipe(
+      switchMap((params) => {
+        this.viewId = params['viewId'];
+        return this.vmMapsQuery.getByViewId(this.viewId);
+      })
+    );
+
+    this.vmsFiltered = this.VmAkitaService.GetViewVms(true, false);
+
+    this.form
```
Comment (sei-noconnor) [inline src/app/components/map/add-point/add-point.component.ts:73]: Duplicate code? looks like your doing the exact same thing twice.

File: src/app/components/map/add-point/add-point.component.ts [hunk 5]
```diff
@@ -28,67 +34,97 @@ export class AddPointComponent implements OnInit {
...
-    this.getVms();
-    this.getVmMaps();
+    this.vmMapsFiltered = this.route.params.pipe(
+      switchMap((params) => {
+        this.viewId = params['viewId'];
+        return this.vmMapsQuery.getByViewId(this.viewId);
+      })
+    );
+
+    this.vmsFiltered = this.VmAkitaService.GetViewVms(true, false);
```
Comment (sei-noconnor) [inline src/app/components/map/add-point/add-point.component.ts:71]: As a general rule we want to use the the akita service to update/change the state and subscribe to those changes through queries in the components. What's the Justification for using the service directly to assign a component value, as opposed to using the query?

**Review by sei-noconnor — APPROVED** [review]

**Review by sei-chershberger — APPROVED** [review]

**Review by sei-aschlackman — APPROVED** [review]

**Review by sei-ebram — APPROVED** [review]

**Pull request merged**